Jenkinsfile

```diff
@@ -53,7 +41,27 @@ node {
       }
     } else {
       stage('Helm install') {
-        defraUtils.deployChart(kubeCredsId, registry, imageName, containerTag, extraCommands)
+        withCredentials([
```

Comment:
I'm not sure I agree with this approach.

I think you're pulling more out of the shared library, and duplicating code that always needs to be set back into the Jenkins file.

Another approach could be to pass in to a library "buildExtraCommands" method a parameter array that would build up the string, and could set the values file and the redeployOnChange value. That would remove duplication from the Jenkinsfile and put it into the library.

Reply:
I haven't pulled anything out of the shared library.

I think there's a small amount of duplication that could be moved to a shared library function but I don't think `buildExtraCommands()` is a good function to have as it's unclear what it should be used for. Perhaps we could look at adding some functionality to the `deployChart` function so it can receive Helm values and build the parameter string, which would seem more clear than passing a generic `extraCommands` argument. However, I think that should be part of the general library refactor so I'd prefer to add a note to that ticket rather than refactor the library as part of this work.

Reply:
I agree.